src/org/labkey/test/components/bootstrap/LoggedOutDialog.java

```diff
@@ -0,0 +1,98 @@
+package org.labkey.test.components.bootstrap;
+
+
+import org.labkey.test.Locator;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+
+public class LoggedOutDialog extends ModalDialog
+{
+    protected LoggedOutDialog(WebDriver driver)
+    {
+        super(new ModalDialogFinder(driver).withTitle("Logged Out")
+                .waitFor().getComponentElement(), driver);
+    }
+
+    static public LoggedOutDialog waitFor(WebDriver driver)
+    {
+        return new LoggedOutDialog(driver);
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public void clickReloadPage()
+    {
+        elementCache().reloadPageBtn.click();
+    }
+
+    /**
+     *  Checks the background/masked elements behind the dialog to ensure that the expected content
+     *  areas are blurred/styled as expected.
+     *  Because not all of these page sections are present in all situations (notably, in apps like
+     *  SM, FM, Biologics) this checks for the ones present, and if any of them are both present and
+     *  not styled as expected, will return false.
+     * @return True if all page sections that are present are styled with 'lk-content-blur', otherwise false
+     */
+    public boolean isContentBlurred()
+    {
+        List<Boolean> factors = new ArrayList<>();
+        Boolean headerBlurred = null;
+        var lkHeader = elementCache().lkHeader.findOptionalElement(getDriver());
+        if (lkHeader.isPresent())
+            factors.add(lkHeader.get().getAttribute("class").contains("lk-content-blur"));
+
+        Boolean bodyBlurred = null;
+        var lkBody = elementCache().lkBody.findOptionalElement(getDriver());
+        if (lkBody.isPresent())
+            factors.add(lkBody.get().getAttribute("class").contains("lk-content-blur"));
+
+        Boolean footerBlurred = null;
+        var footerBlock = elementCache().footerBlock.findOptionalElement(getDriver());
+        if (footerBlock.isPresent())
+            factors.add(footerBlock.get().getAttribute("class").contains("lk-content-blur"));
+
+        // if any of these bodyParts are present and don't have the lk-content-blur style applied, return false
+        // otherwise, true
+        for(Boolean fact : factors)
+        {
+            if(!fact)
+                return false;
+        }
+        return true;
+    }
```

Comment:
Interesting approach to put this check with dialog, it does make it easy to find.

Is there a way to check that this is the top level dialog/component? The scenario: a page has a dialog open then gets the timeout / logged out prompt. This blurred should be rendered on top of that dialog.

Reply:
I haven't come up with a universal way of identifying whatever the top-level component is.
The top of the DOM hierarchy (the 'body' element under /html) will typically get a style to note that it is under a modal dialog when it is, but I haven't seen this happen when the top element is something like a browser alert or a non-bootstrap dialog.

I've focused on the lk-header-ct/lk-body-ct/footer elements because they are the ones we style behind the dialog, not as a guarantee future labkey won't surface stuff that isn't contained in them